engine.go

```diff
@@ -1,16 +1,121 @@
 package fuego
 
-func NewEngine() *Engine {
-	return &Engine{
-		OpenAPI:      NewOpenAPI(),
-		ErrorHandler: ErrorHandler,
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"log/slog"
+	"os"
+	"path/filepath"
+)
+
+func NewEngine(options ...func(*Engine)) *Engine {
+	e := &Engine{
+		OpenAPI:       NewOpenAPI(),
+		OpenAPIConfig: defaultOpenAPIConfig,
+		ErrorHandler:  ErrorHandler,
+	}
+	for _, option := range options {
+		option(e)
 	}
+	return e
 }
 
 // The Engine is the main struct of the framework.
 type Engine struct {
-	OpenAPI      *OpenAPI
-	ErrorHandler func(error) error
+	OpenAPI       *OpenAPI
+	ErrorHandler  func(error) error
+	OpenAPIConfig OpenAPIConfig
 
 	acceptedContentTypes []string
 }
+
+type OpenAPIConfig struct {
```

Comment:
One commit has a typo

495aa54a47354777f7bcb026398d8dc518fbea60

![Screenshot_20241225_071157_GitHub.jpg](https://github.com/user-attachments/assets/a8ee564f-5a79-40f8-9b1f-75a158d915e5)

Reply:
Ah, thanks I’ll get it in the morning. Really need to stop pushing so late